pyproject.toml

```diff
@@ -53,18 +60,35 @@ opentracing = {version = "*", optional = true}
 py-zipkin = {version = "*", optional = true}
 jaeger-client = {version = "*", optional = true}
 prometheus-client = {version = "*", optional = true}
+recommonmark = {version = "*", optional = true}
+Sphinx = {version = "<=3.5.4", optional = true}
+sphinx-click = {version = "*", optional = true}
+sphinx-rtd-theme = {version = "*", optional = true}
+sphinxcontrib-fulltoc = {version = "*", optional = true}
+sphinxcontrib-spelling = {version = "*", optional = true}
+sphinx-copybutton = {version = "*", optional = true}
+pyenchant = {version = "*", optional = true}
+types-click = {version = "*", optional = true}
+types-chardet = {version = "*", optional = true}
+types-setuptools = {version = "*", optional = true}
+types-PyYAML = {version = "*", optional = true}
+pyspark-stubs = {version = "*", optional = true}
+types-filelock = {version = "*", optional = true}
+
+[tool.poetry.extras]
+model-server = ["prometheus-client", "jaeger-client", "py-zipkin", "opentracing"]
+docs = ["recommonmark", "sphinx", "sphinx-click", "sphinx_rtd_theme", "sphinxcontrib-fulltoc", "sphinxcontrib-spelling", "sphinx_copybutton", "pyenchant"]
+types = ["types-click", "types-chardet", "types-setuptools", "pyspark-stubs", "types-filelock", "types-PyYAML"]
```

Comment:
I think types are pretty essential to the dev env and should be included by default.

Reply:
But then the package will become a lot heavier. I think we should keep the behaviour from setup.py

Reply:
Do packages contain dev dependencies by default?

Reply:
yes